client/client-multi/client-c/CMakeLists.txt

```diff
@@ -278,11 +279,14 @@ add_subdirectory(Modules/cppcheck)
 #Use application for another platform
 if(${KAA_PLATFORM} STREQUAL "posix")
     option(KAA_RUNTIME_KEY_GENERATION "Enable RSA key generation at runtime" ON)
-    if(${KAA_RUNTIME_KEY_GENERATION})
+endif()
+
+if(${KAA_RUNTIME_KEY_GENERATION})
+    if(${KAA_PLATFORM} STREQUAL "posix")
```

Comment:
To prevent double explansion, use `KAA_PLATFORM` instead of `${KAA_PLATFORM}`

Consider following code:
```cmake

set(FOO "posix")

if(PLATFORM STREQUAL "posix")
	message("posix")
else()
	message("non-posix")
endif()

if(${PLATFORM} STREQUAL "posix")
	message("posix")
else()
	message("non-posix")
endif()

if("${PLATFORM}" STREQUAL "posix")
	message("posix")
else()
	message("non-posix")
endif()

if("${PLATFORM} " STREQUAL "posix ")
	message("posix")
else()
	message("non-posix")
endif()
```

Running it with `cmake -DPLATFORM=FOO -P example.cmake` produces following output:
```
non-posix
posix
posix
non-posix
```

See also [SO thread](http://stackoverflow.com/questions/19982340/cmake-compare-to-empty-string-with-strequal-failed).